File: hw1/hw1.java
import java.util.Scanner;


class stack {
    char[] stck = new char[100];
    int top;
    public void initialize() {
        top = -1;
    }
    public void push(char ch) {
        stck[++top] = ch;
    }
    public char pop() {
        char tmp = stck[top--];
        return tmp;
    }
}
public class hw1 {
    
    public static void main(String[] args) {
        
        for (;;) {

            String line;
            stack new_stack = new stack();
            new_stack.initialize();

            Scanner input = new Scanner(System.in);
            System.out.print("enter your expression:");
            line = input.nextLine();
            //char[] line_array = new char[line.length()];

            for (int i = 0; i < line.length(); i++) 
            {
                char pop_left, pop_right, tmp_num;
                if (line.charAt(i) <= '9' && line.charAt(i) >= '1')
                    new_stack.push((char)(line.charAt(i) - '0'));
                else
                {
                    switch (line.charAt(i)) {
                        case '+':
                            pop_right = new_stack.pop();
                            pop_left = new_stack.pop();
                            new_stack.push((char)(pop_left + pop_right));
                            break;
                        case '-':
                            pop_right = new_stack.pop();
                            pop_left = new_stack.pop();
                            new_stack.push((char)(pop_left - pop_right));
                            break;
                        case '*':
                            pop_right = new_stack.pop();
                            pop_left = new_stack.pop();
                            new_stack.push((char)(pop_left * pop_right));
                            break;
                        case '/':
                            pop_right = new_stack.pop();
                            pop_left = new_stack.pop();
                            new_stack.push((char)(pop_left / pop_right));
                            break;
                        case '%':
                            pop_right = new_stack.pop();
                            pop_left = new_stack.pop();
                            new_stack.push((char)(pop_left % pop_right));
                            break;
                        case '^':
                            pop_right = new_stack.pop();
                            pop_left = new_stack.pop();
                            new_stack.push((char)Math.pow(pop_left, pop_right));
                            break;
                        default:
                            break;
                    }
                }
            }
            int ans = new_stack.pop();
            System.out.println("the value of expression is:" + ans);
            loops:
                for (;;) {
                    System.out.println("continue?");
                    char opt;
                    Scanner optScanner = new Scanner(System.in);
                    opt = optScanner.next().charAt(0);
                    if (opt == 'Y' || opt == 'y')
                        break loops;
                    else if (opt == 'N' || opt == 'n')
                        System.exit(0);//end the program
                    else {
                        System.out.println("enter Y/y or N/n");
                        continue loops;
                    }
                }
        }
    }
}

File: hw1/hw1.py
import math
class Stack:
    def __init__(self):
        self.stck = []
        self.top = -1
    def initialize(self):
        self.top = -1
    def push(self, ch):
        self.top += 1
        self.stck.append(int(ch))
    def Pop(self):
        #tmp = self.stck[self.top]
        #self.top -= 1
        tmp = self.stck.pop()
        self.top -= 1
        return tmp
    
#main
while True:
    new_stack = Stack()
    new_stack.initialize()

    line = input("enter your expression:")
    for i in range(len(line)):
        if line[i].isdigit():
            new_stack.push(line[i])
        else:
            if line[i] == "+":
                pop_right = new_stack.Pop()
                pop_left = new_stack.Pop()
                new_stack.push(int(pop_right + pop_left))
            elif line[i] == "-":
                pop_right = new_stack.Pop()
                pop_left = new_stack.Pop()
                new_stack.push(int(pop_left - pop_right))
            elif line[i] == "*":
                pop_right = new_stack.Pop()
                pop_left = new_stack.Pop()
                new_stack.push(int(pop_left * pop_right))
            elif line[i] == "/":
                pop_right = new_stack.Pop()
                pop_left = new_stack.Pop()
                new_stack.push(int(pop_left / pop_right))
            elif line[i] == "%":
                pop_right = new_stack.Pop()
                pop_left = new_stack.Pop()
                new_stack.push(int(pop_left % pop_right))
            elif line[i] == "^":
                pop_right = new_stack.Pop()
                pop_left = new_stack.Pop()
                new_stack.push(math.pow(pop_left, pop_right))
        
    
    ans = new_stack.Pop()
    print("the value if the expression is:", int(ans))

    while True:
        opt = input("continue?(Y,y/N,n)")
        if opt == "Y" or opt == "y":
            break
        elif opt == "N" or opt =="n":
            exit(0)
        else:
            print("enter Y/y or N/n")